evalml/utils/gen_utils.py

```diff
@@ -242,7 +242,20 @@ def get_importable_subclasses(base_class, used_in_automl=True):
                 f"Could not import class {cls.__name__} in get_importable_subclasses"
             )
     if used_in_automl:
-        classes = [cls for cls in classes if cls.__name__ not in _not_used_in_automl]
+        from evalml.pipelines import Transformer
+        from evalml.problem_types import ProblemTypes
+
+        classes = [
+            cls
+            for cls in classes
+            if (
+                cls.__name__ not in _not_used_in_automl
+                and (
+                    issubclass(cls, Transformer)
```

Comment:
What's the purpose of `issubclass(cls, Transformer)`?

Reply:
The classes iterated through here can be both estimators and transformers. Estimators have a `supported_problem_types` attribute, but Transformers don't. If we try to check a transformer's supported_problem_types, we just get an error, so we have to check if it's a Transformer first. (I learned this the hard way 😅

Reply:
I think we can just add the clusterers to `_not_used_in_automl` ?

Reply:
I started out adding each clusterer to `_not_used_in_automl` as I added it, but it became a pain to remember to add, so I changed it. My thinking here is that this way we ensure no clusterers, present or future, are included in search. I'm happy to change it back, but personally I'm a fan of the less hard-coded approach.

Reply:
Good point! But I think there are better ways of accomplishing that - maybe we can add an `used_in_search` attribute to component classes and then update this function to respect that value. Then you could just set `used_in_search` in the clusterer base class to `False`.

I think we should stick with the current pattern of modifying `_not_used_in_automl` and tackle ways to improve this method in a separate issue.